### PR TITLE
fixed timer

### PR DESCRIPTION
**Pull request by willharrison**
## `calculateElapsedTime()` explaination

* First we store the elapsed time between the beginning of the game and the current time. `Line 73`
* We then subtract the duration between each pair of pause and resume from the elapsed time. `75 - 80`

### CODE REVIEW

**Comment by ALSchwalm** [comment]
The waiting quad does not seem to drop on its own (after 4 seconds has elapsed) for the first quad.

**Comment by willharrison** [comment]
ah, I didn't notice that.  I will fix it.

**Comment by willharrison** [comment]
try now

**Comment by willharrison** [comment]
fixed

**Comment by ALSchwalm** [comment]
This seems to be working, but I'm somewhat confused by the logic of it. I'll probably look at this more tomorrow morning and merge it then.

**Comment by willharrison** [comment]
I noticed you were using 3000 for the transitions so I just gave the timer -3000 for a buffer.

File: js/app/state/create.js [hunk 1]
```diff
@@ -16,19 +16,29 @@ function(background, music){
...
         game.load.audio('destroy', 'assets/sounds/destroy.wav');
         game.load.audio('move', 'assets/sounds/move.wav');
         game.load.audio('rotate', 'assets/sounds/rotate.wav');
         game.load.start();
 
+        game.onPause.add(function() {
+            timer.pause();
+        });
+
+        game.onResume.add(function() {
```
Comment (ALSchwalm) [inline js/app/state/create.js]: Please put both of these in Timer.init

**Comment by ALSchwalm** [comment]
I just don't follow why the array is necessary. Can you add a description of the algorithm to this PR?

**Comment by willharrison** [comment]
I put up a little explanation.

**Comment by ALSchwalm** [comment]
Based on your explanation, why not store only the starting time and the cumulative difference? That is, every time there is a pause, store that time. Then when there is a resume, add the difference between that time and the current time to a 'totalPausedTime' variable.

**Comment by willharrison** [comment]
That might work but this already works so I don't see a reason to change it.

**Comment by ALSchwalm** [comment]
Here is a diff of this branch without the arrays: [link](https://github.com/ALSchwalm/Quad/commit/9171365004946f2074066ced3dec592268a268fa) It seems to work.

**Comment by ALSchwalm** [comment]
lgtm, merging

**Pull request merged**